# Luxury Watch Brand Analysis and Hierarchy Establishment
Taking an in depth look on a luxury watch dataset containing over 280,000 records. This project will compare brands and their models to their materials and pricing in an effort to establish a luxury watch brand hierarchy. Perhaps consumer purchasing trends can be predicted through a machine learning model as well.

In [110]:
# import dependencies
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [111]:
# load in the csv file and create a DataFrame finding number of rows
watch_df = pd.read_csv("Resources/Watches.csv", low_memory=False)
watch_df.tail()

Unnamed: 0                                               name  \
284486        6438  Zenith Defy El Primero\n21 TITANIUM 95.9000.90...   
284487        6439  Zenith Chronomaster Sport\n41mm White 03.3100....   
284488        6440   Zenith El Primero\n50th Anniversary A386 Limited   
284489        6441  Zenith Chronomaster Sport\nWhite Dial Chronogr...   
284490        6442  Zenith El Primero Chronomaster\n03.2040.400/69...   

          price   brand                    model                    ref  \
284486   $9,790  Zenith          Defy El Primero  95.9000.9004/78.M9000   
284487   $8,450  Zenith       Chronomaster Sport  03.3100.3600/69.M3100   
284488  $16,500  Zenith               El Primero    30.A386.400/69.C807   
284489   $9,000  Zenith       Chronomaster Sport  03.3100.3600/69.M3100   
284490   $6,833  Zenith  El Primero Chronomaster                    NaN   

             mvmt  casem   bracem                   yop       cond  \
284486        NaN    NaN      NaN                  2022  Very good   
284487        NaN    NaN      NaN  2021 (Approximation)  Very good   
284488        NaN    NaN      NaN                  2019  Very good   
284489        NaN    NaN      NaN                  2021     Unworn   
284490  Automatic  Steel  Leather                  2019  Very good   

                       sex   size condition  
284486  Men's watch/Unisex  44 mm       NaN  
284487  Men's watch/Unisex  41 mm       NaN  
284488  Men's watch/Unisex  38 mm       NaN  
284489  Men's watch/Unisex  41 mm       NaN  
284490                 NaN  42 mm       NaN

In [112]:
# Check for duplicated rows
duplicated_df = watch_df.duplicated().sum()
duplicated_df

0

In [113]:
# Check to see if "condition" column is an empty colum
watch_df['condition'].isna().sum()

212922

In [114]:
# Check to see if "cond" column has a lot of empty rows
watch_df['cond'].isna().sum()

75987

In [115]:
# Take a closer look at all NaN column counts
nan_count = watch_df.isna().sum()

print(nan_count)

Unnamed: 0         0
name           72585
price            406
brand            131
model          30466
ref            43152
mvmt          196685
casem         164271
bracem        174896
yop              134
cond           75987
sex            95805
size           23597
condition     212922
dtype: int64


In [116]:
# See if merging the "cond" and "condition" columns works
watch_df["Condition"] = watch_df["cond"].fillna(watch_df["condition"])
watch_df.tail()

Unnamed: 0                                               name  \
284486        6438  Zenith Defy El Primero\n21 TITANIUM 95.9000.90...   
284487        6439  Zenith Chronomaster Sport\n41mm White 03.3100....   
284488        6440   Zenith El Primero\n50th Anniversary A386 Limited   
284489        6441  Zenith Chronomaster Sport\nWhite Dial Chronogr...   
284490        6442  Zenith El Primero Chronomaster\n03.2040.400/69...   

          price   brand                    model                    ref  \
284486   $9,790  Zenith          Defy El Primero  95.9000.9004/78.M9000   
284487   $8,450  Zenith       Chronomaster Sport  03.3100.3600/69.M3100   
284488  $16,500  Zenith               El Primero    30.A386.400/69.C807   
284489   $9,000  Zenith       Chronomaster Sport  03.3100.3600/69.M3100   
284490   $6,833  Zenith  El Primero Chronomaster                    NaN   

             mvmt  casem   bracem                   yop       cond  \
284486        NaN    NaN      NaN                  2022  Very good   
284487        NaN    NaN      NaN  2021 (Approximation)  Very good   
284488        NaN    NaN      NaN                  2019  Very good   
284489        NaN    NaN      NaN                  2021     Unworn   
284490  Automatic  Steel  Leather                  2019  Very good   

                       sex   size condition  Condition  
284486  Men's watch/Unisex  44 mm       NaN  Very good  
284487  Men's watch/Unisex  41 mm       NaN  Very good  
284488  Men's watch/Unisex  38 mm       NaN  Very good  
284489  Men's watch/Unisex  41 mm       NaN     Unworn  
284490                 NaN  42 mm       NaN  Very good

In [117]:
# check number of empty rows in "Condition" column
watch_df["Condition"].isna().sum()

4418

In [118]:
# delete "cond" and "condition" columns
watch_df = watch_df.drop(columns=['cond', 'condition'])

In [119]:
# rename columns to clean headers
watch_df.rename(columns={'name': 'Name', 'price': 'Price', 'brand': 'Brand', 'model': 'Model',
                         'ref': 'Reference', 'mvmt': 'Auto/Battery Powered', 'casem': 'Case Material', 'bracem': 'Brace Material',
                         'yop': 'Year of Production','sex':'Sex','size':'Size'}, inplace=True)
watch_df

Unnamed: 0                                               Name  \
0                0  Audemars Piguet Royal Oak Offshore Chronograph...   
1                1  Audemars Piguet Royal Oak Selfwinding\n39mm Bl...   
2                2  Audemars Piguet Royal Oak Chronograph\nBlue Di...   
3                3  Audemars Piguet Royal Oak Chronograph\nSelfwin...   
4                4  Audemars Piguet Royal Oak Offshore Chronograph...   
...            ...                                                ...   
284486        6438  Zenith Defy El Primero\n21 TITANIUM 95.9000.90...   
284487        6439  Zenith Chronomaster Sport\n41mm White 03.3100....   
284488        6440   Zenith El Primero\n50th Anniversary A386 Limited   
284489        6441  Zenith Chronomaster Sport\nWhite Dial Chronogr...   
284490        6442  Zenith El Primero Chronomaster\n03.2040.400/69...   

           Price            Brand                           Model  \
0        $43,500  Audemars Piguet  Royal Oak Offshore Chronograph   
1        $71,500  Audemars Piguet           Royal Oak Selfwinding   
2        $79,191  Audemars Piguet           Royal Oak Chronograph   
3       $108,000  Audemars Piguet           Royal Oak Chronograph   
4        $27,500  Audemars Piguet  Royal Oak Offshore Chronograph   
...          ...              ...                             ...   
284486    $9,790           Zenith                 Defy El Primero   
284487    $8,450           Zenith              Chronomaster Sport   
284488   $16,500           Zenith                      El Primero   
284489    $9,000           Zenith              Chronomaster Sport   
284490    $6,833           Zenith         El Primero Chronomaster   

                    Reference Auto/Battery Powered Case Material  \
0        26237ST.OO.1000ST.01                  NaN           NaN   
1        15300ST.OO.1220ST.02                  NaN           NaN   
2                     26331ST            Automatic         Steel   
3        26715ST.OO.1356ST.01            Automatic         Steel   
4        26170ST.OO.1000ST.01            Automatic         Steel   
...                       ...                  ...           ...   
284486  95.9000.9004/78.M9000                  NaN           NaN   
284487  03.3100.3600/69.M3100                  NaN           NaN   
284488    30.A386.400/69.C807                  NaN           NaN   
284489  03.3100.3600/69.M3100                  NaN           NaN   
284490                    NaN            Automatic         Steel   

       Brace Material    Year of Production                 Sex        Size  \
0                 NaN                  2019  Men's watch/Unisex       42 mm   
1                 NaN                  2012  Men's watch/Unisex       39 mm   
2               Steel               Unknown                 NaN       41 mm   
3               Steel  2022 (Approximation)  Men's watch/Unisex       38 mm   
4               Steel               Unknown  Men's watch/Unisex  42 x 54 mm   
...               ...                   ...                 ...         ...   
284486            NaN                  2022  Men's watch/Unisex       44 mm   
284487            NaN  2021 (Approximation)  Men's watch/Unisex       41 mm   
284488            NaN                  2019  Men's watch/Unisex       38 mm   
284489            NaN                  2021  Men's watch/Unisex       41 mm   
284490        Leather                  2019                 NaN       42 mm   

        Condition  
0          Unworn  
1       Very good  
2          Unworn  
3             New  
4       Very good  
...           ...  
284486  Very good  
284487  Very good  
284488  Very good  
284489     Unworn  
284490  Very good  

[284491 rows x 13 columns]

In [120]:
# delete "Unnamed:0" column
watch_df = watch_df.drop(columns=['Unnamed: 0'])
watch_df.head()

Name     Price  \
0  Audemars Piguet Royal Oak Offshore Chronograph...   $43,500   
1  Audemars Piguet Royal Oak Selfwinding\n39mm Bl...   $71,500   
2  Audemars Piguet Royal Oak Chronograph\nBlue Di...   $79,191   
3  Audemars Piguet Royal Oak Chronograph\nSelfwin...  $108,000   
4  Audemars Piguet Royal Oak Offshore Chronograph...   $27,500   

             Brand                           Model             Reference  \
0  Audemars Piguet  Royal Oak Offshore Chronograph  26237ST.OO.1000ST.01   
1  Audemars Piguet           Royal Oak Selfwinding  15300ST.OO.1220ST.02   
2  Audemars Piguet           Royal Oak Chronograph               26331ST   
3  Audemars Piguet           Royal Oak Chronograph  26715ST.OO.1356ST.01   
4  Audemars Piguet  Royal Oak Offshore Chronograph  26170ST.OO.1000ST.01   

  Auto/Battery Powered Case Material Brace Material    Year of Production  \
0                  NaN           NaN            NaN                  2019   
1                  NaN           NaN            NaN                  2012   
2            Automatic         Steel          Steel               Unknown   
3            Automatic         Steel          Steel  2022 (Approximation)   
4            Automatic         Steel          Steel               Unknown   

                  Sex        Size  Condition  
0  Men's watch/Unisex       42 mm     Unworn  
1  Men's watch/Unisex       39 mm  Very good  
2                 NaN       41 mm     Unworn  
3  Men's watch/Unisex       38 mm        New  
4  Men's watch/Unisex  42 x 54 mm  Very good

In [121]:
# Convert "Unknown" values in "Year of Production" column to NaN
watch_df['Year of Production']=watch_df['Year of Production'].replace(to_replace='Unknown', value=np.nan)
watch_df.head()

Name     Price  \
0  Audemars Piguet Royal Oak Offshore Chronograph...   $43,500   
1  Audemars Piguet Royal Oak Selfwinding\n39mm Bl...   $71,500   
2  Audemars Piguet Royal Oak Chronograph\nBlue Di...   $79,191   
3  Audemars Piguet Royal Oak Chronograph\nSelfwin...  $108,000   
4  Audemars Piguet Royal Oak Offshore Chronograph...   $27,500   

             Brand                           Model             Reference  \
0  Audemars Piguet  Royal Oak Offshore Chronograph  26237ST.OO.1000ST.01   
1  Audemars Piguet           Royal Oak Selfwinding  15300ST.OO.1220ST.02   
2  Audemars Piguet           Royal Oak Chronograph               26331ST   
3  Audemars Piguet           Royal Oak Chronograph  26715ST.OO.1356ST.01   
4  Audemars Piguet  Royal Oak Offshore Chronograph  26170ST.OO.1000ST.01   

  Auto/Battery Powered Case Material Brace Material    Year of Production  \
0                  NaN           NaN            NaN                  2019   
1                  NaN           NaN            NaN                  2012   
2            Automatic         Steel          Steel                   NaN   
3            Automatic         Steel          Steel  2022 (Approximation)   
4            Automatic         Steel          Steel                   NaN   

                  Sex        Size  Condition  
0  Men's watch/Unisex       42 mm     Unworn  
1  Men's watch/Unisex       39 mm  Very good  
2                 NaN       41 mm     Unworn  
3  Men's watch/Unisex       38 mm        New  
4  Men's watch/Unisex  42 x 54 mm  Very good

In [122]:
# Delete "(Approximation)" values in "Year of Production" column
watch_df['Year of Production'] = watch_df['Year of Production'].replace('[(Approximation)]', '', regex=True)
watch_df.tail()

Name    Price   Brand  \
284486  Zenith Defy El Primero\n21 TITANIUM 95.9000.90...   $9,790  Zenith   
284487  Zenith Chronomaster Sport\n41mm White 03.3100....   $8,450  Zenith   
284488   Zenith El Primero\n50th Anniversary A386 Limited  $16,500  Zenith   
284489  Zenith Chronomaster Sport\nWhite Dial Chronogr...   $9,000  Zenith   
284490  Zenith El Primero Chronomaster\n03.2040.400/69...   $6,833  Zenith   

                          Model              Reference Auto/Battery Powered  \
284486          Defy El Primero  95.9000.9004/78.M9000                  NaN   
284487       Chronomaster Sport  03.3100.3600/69.M3100                  NaN   
284488               El Primero    30.A386.400/69.C807                  NaN   
284489       Chronomaster Sport  03.3100.3600/69.M3100                  NaN   
284490  El Primero Chronomaster                    NaN            Automatic   

       Case Material Brace Material Year of Production                 Sex  \
284486           NaN            NaN               2022  Men's watch/Unisex   
284487           NaN            NaN              2021   Men's watch/Unisex   
284488           NaN            NaN               2019  Men's watch/Unisex   
284489           NaN            NaN               2021  Men's watch/Unisex   
284490         Steel        Leather               2019                 NaN   

         Size  Condition  
284486  44 mm  Very good  
284487  41 mm  Very good  
284488  38 mm  Very good  
284489  41 mm     Unworn  
284490  42 mm  Very good

In [123]:
# Review all NaN column counts
nan_count = watch_df.isna().sum()
print(nan_count)

Name                     72585
Price                      406
Brand                      131
Model                    30466
Reference                43152
Auto/Battery Powered    196685
Case Material           164271
Brace Material          174896
Year of Production       96091
Sex                      95805
Size                     23597
Condition                 4418
dtype: int64


In [124]:
# drop NaN values, reset index, and put into new dataframe
clean_df = watch_df.dropna()
clean_df = clean_df.reset_index(drop=True)
clean_df.head()

Name             Price  \
0  Audemars Piguet Royal Oak Chronograph\nSelfwin...          $108,000   
1  Audemars Piguet Royal Oak Offshore Diver\n2018...           $41,000   
2  Audemars Piguet Royal Oak Offshore Diver Chron...           $43,500   
3  Audemars Piguet Royal Oak Offshore Chronograph...           $37,800   
4  Audemars Piguet Royal Oak Chronograph\nSelfwin...  Price on request   

             Brand                                 Model  \
0  Audemars Piguet                 Royal Oak Chronograph   
1  Audemars Piguet              Royal Oak Offshore Diver   
2  Audemars Piguet  Royal Oak Offshore Diver Chronograph   
3  Audemars Piguet        Royal Oak Offshore Chronograph   
4  Audemars Piguet                 Royal Oak Chronograph   

              Reference Auto/Battery Powered Case Material Brace Material  \
0  26715ST.OO.1356ST.01            Automatic         Steel          Steel   
1  15710ST.OO.A051CA.01            Automatic         Steel         Rubber   
2  26703ST.OO.A051CA.01            Automatic         Steel         Rubber   
3  26470IO.OO.A006CA.01            Automatic      Titanium         Rubber   
4  26715ST.OO.1356ST.02            Automatic         Steel          Steel   

  Year of Production                 Sex   Size  Condition  
0              2022   Men's watch/Unisex  38 mm        New  
1              2018   Men's watch/Unisex  42 mm     Unworn  
2              2016   Men's watch/Unisex  42 mm  Very good  
3               2019  Men's watch/Unisex  42 mm  Very good  
4               2023  Men's watch/Unisex  38 mm     Unworn

In [142]:
# Review all NaN column counts
nan_count = clean_df.isna().sum()
print(nan_count)
length = len(clean_df)
print(f'Clean DataFrame Record Count: {length}')

Name                    0
Price                   0
Brand                   0
Model                   0
Reference               0
Auto/Battery Powered    0
Case Material           0
Brace Material          0
Year of Production      0
Sex                     0
Size                    0
Condition               0
dtype: int64
Clean DataFrame Record Count: 34541


In [163]:
# Save DataFrame to working directory as csv file for visualization in Tableau
clean_df.to_csv('Resources/cleaned_watch.csv')

## Cleaning Analysis
 number of records in watch_df: 284491
 number of records in clean_df: 34541 
 percent data retention: 12.1%
 drop_na plan: FAILURE
 NEXT STEP: Divide watch_df into separate dataframes, clean individually, export as separate csv files, Tableau
 DATAFRAMES: Should i include name? Removing the column from select dataframes will increase record retention
 Brand_df: {Price, Brand, Model, Year of Production, Size, Condition}
 Gender_df: {Price, Brand, Model, Year of Production, Sex, Size, Condition}
 Material_df: {Price, Brand, Model, Year of Production, Case Material, Brace Material, Condition}
 Gender_material_df: {Year of Production, Case Material, Brace Material, Sex, Condition}
 

## Brand Analysis:
Columns:{Price, Brand, Model, Year of Production, Size, Condition}
Goal: Determine the most popular brands, watch models, and sizes. What are the brands/models in the best conditions?

In [144]:
# Create Brand_df with specified columns
Brand_df = watch_df[['Price','Brand','Model','Year of Production','Size','Condition']].copy()
Brand_df.tail()

Price   Brand                    Model Year of Production   Size  \
284486   $9,790  Zenith          Defy El Primero               2022  44 mm   
284487   $8,450  Zenith       Chronomaster Sport              2021   41 mm   
284488  $16,500  Zenith               El Primero               2019  38 mm   
284489   $9,000  Zenith       Chronomaster Sport               2021  41 mm   
284490   $6,833  Zenith  El Primero Chronomaster               2019  42 mm   

        Condition  
284486  Very good  
284487  Very good  
284488  Very good  
284489     Unworn  
284490  Very good

In [145]:
# Clean DataFrame of all NaN values and reset index
Brand_df = Brand_df.dropna()
Brand_df = Brand_df.reset_index(drop=True)
Brand_df.tail()

Price   Brand                    Model Year of Production   Size  \
159481   $9,790  Zenith          Defy El Primero               2022  44 mm   
159482   $8,450  Zenith       Chronomaster Sport              2021   41 mm   
159483  $16,500  Zenith               El Primero               2019  38 mm   
159484   $9,000  Zenith       Chronomaster Sport               2021  41 mm   
159485   $6,833  Zenith  El Primero Chronomaster               2019  42 mm   

        Condition  
159481  Very good  
159482  Very good  
159483  Very good  
159484     Unworn  
159485  Very good

In [157]:
# Check for remaining NaN values
NaN_count = Brand_df.isna().sum()
NaN_count

Price                 0
Brand                 0
Model                 0
Year of Production    0
Size                  0
Condition             0
dtype: int64

In [156]:
length = len(Brand_df)
percent = 159486/284491*100
percentage = round(percent,1)
print(f'Brand Record Length: {length}')
print(f'Record Retention Percentage: {percentage}%')

Brand Record Length: 159486
Record Retention Percentage: 56.1%


In [164]:
# Save to Resources folder as csv file
Brand_df.to_csv('Resources/Brand.csv')

## Gender Analysis:
Columns:{Price, Brand, Model, Year of Production, Sex, Size, Condition}
Goal: Determine the most popular brands, watch models, and sizes amongst genders. Does a certain gender take better care of watches?

In [171]:
# Create Gender_df with specified columns
Gender_df = watch_df[['Price', 'Brand', 'Model', 'Year of Production', 'Sex', 'Size', 'Condition']].copy()
Gender_df = Gender_df.reset_index(drop=True)
Gender_df.tail()

Price   Brand                    Model Year of Production  \
284486   $9,790  Zenith          Defy El Primero               2022   
284487   $8,450  Zenith       Chronomaster Sport              2021    
284488  $16,500  Zenith               El Primero               2019   
284489   $9,000  Zenith       Chronomaster Sport               2021   
284490   $6,833  Zenith  El Primero Chronomaster               2019   

                       Sex   Size  Condition  
284486  Men's watch/Unisex  44 mm  Very good  
284487  Men's watch/Unisex  41 mm  Very good  
284488  Men's watch/Unisex  38 mm  Very good  
284489  Men's watch/Unisex  41 mm     Unworn  
284490                 NaN  42 mm  Very good

In [172]:
# Chech NaN value counts acroos all columns
NaN_count = Gender_df.isna().sum()
NaN_count

Price                   406
Brand                   131
Model                 30466
Year of Production    96091
Sex                   95805
Size                  23597
Condition              4418
dtype: int64

In [173]:
# Clean DataFrame of all NaN values and reset index
Gender_df = Gender_df.dropna()
Gender_df = Gender_df.reset_index(drop=True)
Gender_df.tail()

Price   Brand               Model Year of Production  \
100108   $9,950  Zenith                Defy               2022   
100109   $9,790  Zenith     Defy El Primero               2022   
100110   $8,450  Zenith  Chronomaster Sport              2021    
100111  $16,500  Zenith          El Primero               2019   
100112   $9,000  Zenith  Chronomaster Sport               2021   

                       Sex   Size  Condition  
100108  Men's watch/Unisex  45 mm  Very good  
100109  Men's watch/Unisex  44 mm  Very good  
100110  Men's watch/Unisex  41 mm  Very good  
100111  Men's watch/Unisex  38 mm  Very good  
100112  Men's watch/Unisex  41 mm     Unworn

In [148]:
# Check unique values in 'model' column
unique = len(Gender_df.Model.unique())
unique

863

In [180]:
# Chech NaN value counts acroos all columns
NaN_count = Gender_df.isna().sum()
NaN_count

Price                 0
Brand                 0
Model                 0
Year of Production    0
Sex                   0
Size                  0
Condition             0
dtype: int64

In [181]:
# Check acceptability of cleaning method by analysis of record retention
length = len(Gender_df)
percent = 100113/284491*100
percentage = round(percent,1)
print(f'Gender Record Length: {length}')
print(f'Record Retention Percentage: {percentage}%')

Gender Record Length: 100113
Record Retention Percentage: 35.2%


In [182]:
# Save Gender_df to Resources folder as Gender_Preference.csv
Gender_df.to_csv('Resources/Gender_Preference.csv')

# Material Preference Analysis
Columns: {Price, Brand, Model, Year of Production, Case Material, Brace Material, Condition}
Goal: Provide insight into the trend of preference of watch case and brace material as well as their condition

In [193]:
# Create Material_df with specified columns
Material_df = watch_df[['Price','Year of Production', 'Case Material', 'Brace Material', 'Condition']].copy()
Material_df = Material_df.reset_index(drop=True)
Material_df.tail()

Price Year of Production Case Material Brace Material  Condition
284486   $9,790               2022           NaN            NaN  Very good
284487   $8,450              2021            NaN            NaN  Very good
284488  $16,500               2019           NaN            NaN  Very good
284489   $9,000               2021           NaN            NaN     Unworn
284490   $6,833               2019         Steel        Leather  Very good

In [194]:
# Chech NaN value counts acroos all columns
NaN_count = Material_df.isna().sum()
NaN_count

Price                    406
Year of Production     96091
Case Material         164271
Brace Material        174896
Condition               4418
dtype: int64

In [195]:
# Drop all NaN values across all columns
Material_df = Material_df.dropna()
Material_df = Material_df.reset_index(drop=True)
print(len(Material_df))

75837


In [196]:
# Chech NaN value counts acroos all columns
NaN_count = Gender_df.isna().sum()
NaN_count

Price                 0
Brand                 0
Model                 0
Year of Production    0
Sex                   0
Size                  0
Condition             0
dtype: int64

In [199]:
# Check acceptability of cleaning method by analysis of record retention
length = len(Material_df)
percent = 75811/284491*100
percentage = round(percent,1)
print(f'Materials Record Length: {length}')
print(f'Record Retention Percentage: {percentage}%')

Materials Record Length: 75837
Record Retention Percentage: 26.6%


In [200]:
# Save Material_df to Resources folder as Material_Preference.csv
Material_df.to_csv('Resources/Material_Preference.csv')

# Gender Material Preference Analysis
Columns: {Year of Production, Case Material, Brace Material, Sex, Condition}
Goal: Provide insight into gender preferences of luxury watch case and brace materials as well as their condition

In [203]:
# Create Material_df with specified columns
Gender_Material_df = watch_df[['Year of Production', 'Case Material', 'Brace Material', 'Sex', 'Condition']].copy()
Gender_Material_df = Gender_Material_df.reset_index(drop=True)
Gender_Material_df.tail()

Year of Production Case Material Brace Material                 Sex  \
284486               2022           NaN            NaN  Men's watch/Unisex   
284487              2021            NaN            NaN  Men's watch/Unisex   
284488               2019           NaN            NaN  Men's watch/Unisex   
284489               2021           NaN            NaN  Men's watch/Unisex   
284490               2019         Steel        Leather                 NaN   

        Condition  
284486  Very good  
284487  Very good  
284488  Very good  
284489     Unworn  
284490  Very good

In [205]:
# Chech NaN value counts across all columns
NaN_count = Gender_Material_df.isna().sum()
NaN_count

Year of Production     96091
Case Material         164271
Brace Material        174896
Sex                    95805
Condition               4418
dtype: int64

In [206]:
# Drop all NaN values across all columns
Gender_Material_df = Gender_Material_df.dropna()
Gender_Material_df = Gender_Material_df.reset_index(drop=True)
print(len(Gender_Material_df))

39774


In [207]:
# Chech NaN value counts across all columns
NaN_count = Gender_Material_df.isna().sum()
NaN_count

Year of Production    0
Case Material         0
Brace Material        0
Sex                   0
Condition             0
dtype: int64

In [209]:
# Check acceptability of cleaning method by analysis of record retention
length = len(Gender_Material_df)
percent = 39774/284491*100
percentage = round(percent,1)
print(f'Gender Material Preference Record Length: {length}')
print(f'Record Retention Percentage: {percentage}%')

Gender Material Preference Record Length: 39774
Record Retention Percentage: 14.0%


In [210]:
# Save Gender_Material_df to Resources folder as Gender_Material_Preference.csv
Gender_Material_df.to_csv('Resources/Gender_Material_Preference.csv')